### PR TITLE
PARQUET-297: generate Version class using parquet-generator

### PR DESCRIPTION
**Pull request by kostya-sh**

### CODE REVIEW

File: parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java [hunk 1]
```diff
@@ -256,4 +254,8 @@ private static String align(int value, int digits) {
     result.append(valueString);
     return result.toString();
   }
+  
+  private static int paddedByteCountFromBits(int bitLength) {
```
Comment (rdblue) [inline parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java]: I don't see why these changes should be included in the version pull request. Maybe this was an accident?

File: parquet-generator/src/main/java/org/apache/parquet/version/VersionGenerator.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    add("  public static final String VERSION_NUMBER = \"");
+    add(props.getProperty("versionNumber"));
+    add("\";\n");
+
+    add("  public static final String FULL_VERSION = \"");
+    add(props.getProperty("fullVersion"));
+    add("\";\n");
+
+    add("}\n");
+    writer.close();
```
Comment (rdblue) [inline parquet-generator/src/main/java/org/apache/parquet/version/VersionGenerator.java]: Other than the main method, this looks compatible with the old Version class to me. While it's nit-picking and probably over-cautious, I think we should probably include main.

**Comment by rdblue** [comment]
Thanks for building this @kostya-sh! It looks great, just a couple of comments.

File: parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java (hunk 1 above)
Comment (kostya-sh) [inline parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java]: I had to break circular dependency between parquet-generator and parquet-common. 

This was the only method from parquet-common that was used by parquet-generator, so I've just copied it here and removed dependency on parquet-common (have a look at parquet-generator/pom.xml change).

Comment (rdblue) [inline parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java]: Okay, that makes sense to me. Sounds good.

File: parquet-generator/src/main/java/org/apache/parquet/version/VersionGenerator.java (hunk 2 above)
Comment (kostya-sh) [inline parquet-generator/src/main/java/org/apache/parquet/version/VersionGenerator.java]: Done

File: parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java (hunk 1 above)
Comment (isnotinvain) [inline parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java]: yeah sort of a bummer but not too bad. Can you add a comment to this method: 
```
// duplicated from BytesUtils to avoid a circular dependency between parquet-common and parquet-generator
// TODO: fix the circular dependency by moving the shared bits to a 3rd module (PARQUET-XXXX)
```

**Comment by isnotinvain** [comment]
Overall LGTM, just one minor comment

File: parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java (hunk 1 above)
Comment (rdblue) [inline parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java]: +1

Comment (kostya-sh) [inline parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java]: I will add a comment about ByteUtils. Not sure it is worth fixing though as the method is one-liner, is not parquet specific and easy to understand.

**Comment by isnotinvain** [comment]
+1
